PySight.py

```diff
@@ -477,86 +464,80 @@ def is_map_alert_to_event(p_misp_instance, a_event, a_isight_alert, a_auto_comme
+            new_misp_event.add_tag('veris:action:malware:variety="C2"')
 
         if not (a_isight_alert.url is None):
-            p_misp_instance.add_url(a_event, a_isight_alert.url, 'Network activity', False, "url " + a_auto_comment)
+            new_misp_event.add_attribute(type='url', value=a_isight_alert.url, comment='url {}'.format(a_auto_comment))
 
+        has_email = False
+        new_email_object = MISPObject(name='email', standalone=False)
         # if attack was by E-Mail
         if a_isight_alert.senderAddress:
-            p_misp_instance.add_email_src(a_event, a_isight_alert.senderAddress, False,
-                                          "senderAddress " + a_auto_comment)
+            new_email_object.add_attribute('from', value=a_isight_alert.senderAddress, to_ids=False, comment='senderAddress {}'.format(a_auto_comment))
+            has_email = True
         if a_isight_alert.subject:
-            p_misp_instance.add_email_subject(a_event, a_isight_alert.subject, False,
-                                              "E-mail subject " + a_auto_comment)
-        if a_isight_alert.sourceDomain:
-            p_misp_instance.add_domain(a_event, a_isight_alert.sourceDomain, 'Network activity', False,
-                                       " E-mail source domain " + a_auto_comment)
+            new_email_object.add_attribute('subject', value=a_isight_alert.subject, to_ids=False, comment='E-mail subject {}'.format(a_auto_comment))
+            has_email = True
         if a_isight_alert.senderName:
-            p_misp_instance.add_internal_other(a_event, a_isight_alert.senderName, False,
-                                               "E-mail sender name " + a_auto_comment)
+            new_email_object.add_attribute('from-display-name', value=a_isight_alert.senderName, to_ids=False, comment='E-mail sender name {}'.format(a_auto_comment))
+            has_email = True
+        if a_isight_alert.sourceDomain:
+            attr = new_misp_event.add_attribute(type='domain', value=a_isight_alert.sourceDomain, comment='E-mail source domain {}'.format(a_auto_comment))
+            if has_email:
+                new_email_object.add_reference(attr.uuid, 'related-to', 'E-mail source domain')
         if a_isight_alert.emailLanguage:
-            p_misp_instance.add_internal_other(a_event, a_isight_alert.emailLanguage, False,
-                                               "E-mail language " + a_auto_comment)
-    except TypeError as e:
+            attr = new_misp_event.add_attribute(type='other', value=a_isight_alert.emailLanguage, category='Internal reference', comment='E-mail language {}'.format(a_auto_comment))
+            if has_email:
+                new_email_object.add_reference(attr.uuid, 'related-to', 'E-mail language')
+        if has_email:
+            new_misp_event.add_object(new_email_object)
+        p_misp_instance.add_event(new_misp_event)
```

Comment:
in my version of this revised code when finally sending the event up to the server here, i get this in the `error.log`. where should `info` be set?
```
Error: [PDOException] SQLSTATE[HY000]: General error: 1364 Field 'info' doesn't have a default value Request URL: /events
```

Reply:
That should be fixed in the commit I'll do in a few min.